packages/web-app/src/utils/tokens.ts

```
@@ -147,27 +149,35 @@ export const fetchBalance = async (
   tokenAddress: string,
   ownerAddress: string,
   provider: EthersProviders.Provider,
+  nativeCurrency: NativeTokenData,
   shouldFormat = true
 ) => {
   const contract = new ethers.Contract(tokenAddress, erc20TokenABI, provider);
-  const balance = await contract.balanceOf(ownerAddress);
-
-  if (shouldFormat) {
-    const {decimals} = await getTokenInfo(tokenAddress, provider);
-    return formatUnits(balance, decimals);
+  try {
+    const balance = await contract.balanceOf(ownerAddress);
+
+    if (shouldFormat) {
+      const {decimals} = await getTokenInfo(
+        tokenAddress,
+        provider,
+        nativeCurrency
+      );
+      return formatUnits(balance, decimals);
+    }
+    return balance;
+  } catch (err) {
```

Thanks for adding a try/catch block. But now instead of handling the error we're just swallowing it. I.e., if something goes wrong here we just return `0` and nobody will know why.

Also,  promises come with a "failure state". Unless we explicitly transform the error into a custom error that we can work with further up the call stack, it might not even be worth to do this here.